homeassistant/components/green_planet_energy/sensor.py

```diff
@@ -0,0 +1,260 @@
+        price_key = f"gpe_price_{hour:02d}"
+        if price_key in data:
+            price = data[price_key]
+            if price is not None:
+                prices.append(price)
+
+    return min(prices) if prices else None
+
+
+def get_lowest_price_night(data: dict[str, Any]) -> float | None:
+    """Get the lowest price during night hours (18-6)."""
+    if not data:
+        return None
+
+    prices = []
+    # Evening hours (18-23)
+    for hour in range(18, 24):
+        price_key = f"gpe_price_{hour:02d}"
+        if price_key in data:
+            price = data[price_key]
+            if price is not None:
+                prices.append(price)
+
+    # Early morning hours (0-5)
+    for hour in range(6):
+        price_key = f"gpe_price_{hour:02d}"
+        if price_key in data:
+            price = data[price_key]
+            if price is not None:
+                prices.append(price)
+
+    return min(prices) if prices else None
+
+
+def get_current_price(data: dict[str, Any]) -> float | None:
+    """Get the price for the current hour."""
+    if not data:
+        return None
+
+    current_hour = dt_util.now().hour
+    price_key = f"gpe_price_{current_hour:02d}"
+    return data.get(price_key)
+
+
+@dataclass(frozen=True, kw_only=True)
+class GreenPlanetEnergySensorEntityDescription(SensorEntityDescription):
+    """Describes Green Planet Energy sensor entity."""
+
+    value_fn: Callable[[dict[str, Any]], float | None] | None = None
+
+
+SENSOR_DESCRIPTIONS: list[GreenPlanetEnergySensorEntityDescription] = [
+    # Statistical sensors only - hourly prices available via service
+    GreenPlanetEnergySensorEntityDescription(
+        key="gpe_highest_price_today",
+        translation_key="highest_price_today",
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        device_class=SensorDeviceClass.MONETARY,
+        suggested_display_precision=4,
+        value_fn=get_highest_price_today,
+    ),
+    GreenPlanetEnergySensorEntityDescription(
+        key="gpe_lowest_price_day",
+        translation_key="lowest_price_day",
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        device_class=SensorDeviceClass.MONETARY,
+        suggested_display_precision=4,
+        value_fn=get_lowest_price_day,
+    ),
+    GreenPlanetEnergySensorEntityDescription(
+        key="gpe_lowest_price_night",
+        translation_key="lowest_price_night",
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        device_class=SensorDeviceClass.MONETARY,
+        suggested_display_precision=4,
+        value_fn=get_lowest_price_night,
+    ),
+    GreenPlanetEnergySensorEntityDescription(
+        key="gpe_current_price",
+        translation_key="current_price",
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        device_class=SensorDeviceClass.MONETARY,
```

Comment:
Device class monetary is incorrect, that can only be used for a balance, as eur/kWh is not a balance you can't use it